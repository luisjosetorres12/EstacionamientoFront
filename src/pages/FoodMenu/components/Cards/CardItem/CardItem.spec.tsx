/* eslint-disable react/jsx-props-no-spreading, @typescript-eslint/no-var-requires, global-require */

import React from 'react';
import {
  CardItem,
  CardItemProps,
} from 'pages/FoodMenu/components/Cards/CardItem/index';
import { render, fireEvent } from '@testing-library/react';

jest.mock('react-intersection-observer', () => {
  const { factory } = require('tests/mocks/react-intersection-observer.mock');
  return factory();
});

describe('CardItem Test', () => {
  let props: CardItemProps;

  beforeEach(() => {
    props = {
      buttonText: 'click me',
      cover: 'food.jpg',
      description: 'some description',
      highLight: 'highlight',
      title: 'some title',
      onSelected: jest.fn(),
    };
  });

  it('should match snapshot', () => {
    const { container } = render(<CardItem {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('should call onSelected fn when button is clicked', async () => {
    const { getByText } = render(<CardItem {...props} />);
    const button = getByText(props.buttonText);
    fireEvent.click(button);
    expect(props.onSelected).toHaveBeenCalled();
  });

  it('should call onSelected fn when img is clicked', async () => {
    const { getByTestId } = render(<CardItem {...props} />);
    const button = getByTestId('img-button');
    fireEvent.click(button);
    expect(props.onSelected).toHaveBeenCalled();
  });
});
