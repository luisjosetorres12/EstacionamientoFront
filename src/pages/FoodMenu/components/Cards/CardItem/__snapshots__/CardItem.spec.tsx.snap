// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardItem Test should match snapshot 1`] = `
<div>
  <article
    class="makeStyles-paper-2"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        class="MuiGrid-root makeStyles-imgContainer-1 MuiGrid-item"
      >
        <button
          class="MuiButtonBase-root"
          data-testid="img-button"
          tabindex="0"
          type="button"
        >
          <img
            alt="Food item"
            class="makeStyles-img-3"
            src="/images/food.jpg"
          />
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-direction-xs-column MuiGrid-grid-xs-true"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        >
          <h6
            class="MuiTypography-root makeStyles-title-4 MuiTypography-subtitle1 MuiTypography-gutterBottom"
          >
            some title
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            some description
          </p>
        </div>
        <div
          class="MuiGrid-root makeStyles-priceContainer-5 MuiGrid-item MuiGrid-grid-xs-true"
        >
          <h6
            class="MuiTypography-root makeStyles-highLight-6 MuiTypography-subtitle1"
          >
            highlight
          </h6>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-7 MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              click me
            </span>
          </button>
        </div>
      </div>
    </div>
  </article>
</div>
`;
